src/pybind/mgr/dashboard/services/service.py

```diff
@@ -135,6 +135,23 @@ def restart_rgw_daemons_and_set_credentials(self):
             self.configure_rgw_credentials()
         else:
             logger.error("Not all daemons are up, skipping RGW credentials configuration")
+    
+    def restart_rgw_daemons(self):
+        # Restart RGW daemons
+        logger.info("Restarting RGW daemons")
+        orch = OrchClient.instance()
+        services, _ = orch.services.list(service_type='rgw', offset=0)
+
+        all_daemons_up = True
+        for service in services:
+            logger.info("Verifying service restart for: %s", service['service_id'])
+            daemons_up = verify_service_restart('rgw', service['service_id'])
+            if not daemons_up:
+                logger.error("Service %s restart verification failed", service['service_id'])
+                all_daemons_up = False
+
+        if all_daemons_up:
+            logger.info("All daemons are up")
```

Comment:
to remove the duplicated code, can we reuse this code to the above function as well? so that we don't duplicate codes? we just have to return the all daemons_up and then go on to configure credentials if that is true.

and probably we don't need this
```

        if all_daemons_up:
            logger.info("All daemons are up")
```

```suggestion
        logger.info("Restarting RGW daemons")
        orch = OrchClient.instance()
        services, _ = orch.services.list(service_type='rgw', offset=0)

        all_daemons_up = True
        for service in services:
            logger.info("Verifying service restart for: %s", service['service_id'])
            daemons_up = verify_service_restart('rgw', service['service_id'])
            if not daemons_up:
                logger.error("Service %s restart verification failed", service['service_id'])
                all_daemons_up = False

```